cmake/SDL.cmake

```diff
@@ -101,6 +101,9 @@ if(WIN32)
         # add Clang++ specific flags
         append(CMAKE_EXE_LINKER_FLAGS "-Xlinker /DEPENDENTLOADFLAG:0x2000")
         append(CMAKE_SHARED_LINKER_FLAGS "-Xlinker /DEPENDENTLOADFLAG:0x2000")
+    elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+        # MinGW detected, do not use /DEPENDENTLOADFLAG as it's not supported by MinGW's linker
+        message(STATUS "Skipping /DEPENDENTLOADFLAG for MinGW")
```

Comment:
Would it make sense to use in the first if block:
```
if(UNIX OR MINGW)
```
and in the second if block
```
if(WIN32 AND(NOT MINGW))
```
?

Reply:
The issue isn't the system, ie win32 vs unix, but the compiler type. The parent code block is skipped for unix systems and that's fine, the issue is on win32 **and** the GNU compiler. This is what MingW is.

Basically:

Unix + GNU = not an issue
Unix + not GNU = not an issue
Win32 + not GNU = not an issue
Win32 + GNU = issue

Reply:
How about like:
```
if(WIN32 AND NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
```

Reply:
IIUC the [cmake documentation](https://cmake.org/cmake/help/v3.6/variable/MINGW.html), the `MINGW` variable does exactly what is needed here:
```
Set to true when the compiler is some version of MinGW.
```

Is there a reason why you want to check GNU and not MINGW?

Reply:
None in particular, I just didn't known it existed. Good to know. How about:

```
if(WIN32 AND NOT MINGW)
```

Reply:
Okay, I've squashed and pushed my changes :) 

Thanks @mgouicem !